### PR TITLE
Add support for Extension mechanism

### PR DESCRIPTION
**Pull request by JohnNiang**
#### What type of PR is this?

/kind feature
/area core

#### What this PR does / why we need it:

This PR only adds some basic features for Extension mechanism.

See [Extension Model RFC](https://github.com/halo-dev/rfcs/blob/26768761e69d25e72d57f6c4d1a59f930b71b9d9/extension/README.md) for more.

Something we need to do in next PR:

- Validate Extension.
- Watch Extension change.
- Usage documentation.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

/cc @halo-dev/sig-halo 

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: src/main/java/run/halo/app/extension/Scheme.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package run.halo.app.extension;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import lombok.Data;
```
Comment (ruibaby) [inline src/main/java/run/halo/app/extension/Scheme.java:4]: 似乎没有用到。

Comment (JohnNiang) [inline src/main/java/run/halo/app/extension/Scheme.java:4]:
```suggestion
```

Comment (JohnNiang) [reply src/main/java/run/halo/app/extension/Scheme.java]: Please see https://github.com/halo-dev/halo/pull/2086/commits/22354a7ae3bd5ff834438bb6ffb5ac2b288d7f5f

**Comment by JohnNiang** [comment]
Hi all @halo-dev/sig-halo, Please take some moment to have a look.

**Review by ruibaby — COMMENTED** [review]
/lgtm

unbelievable！

**Pull request merged**